app/backend/src/couchers/servicers/requests.py

```diff
@@ -636,6 +649,31 @@ def MarkLastSeenHostRequest(self, request, context, session):
         session.commit()
         return empty_pb2.Empty()
 
+    def SetHostRequestArchiveStatus(self, request, context, session):
+        host_request = session.execute(
+            select(HostRequest)
+            .where_users_column_visible(context, HostRequest.surfer_user_id)
+            .where_users_column_visible(context, HostRequest.host_user_id)
+            .where(HostRequest.conversation_id == request.host_request_id)
+            .where(or_(HostRequest.surfer_user_id == context.user_id, HostRequest.host_user_id == context.user_id))
+        ).scalar_one_or_none()
+
+        if not host_request:
+            context.abort(grpc.StatusCode.NOT_FOUND, errors.HOST_REQUEST_NOT_FOUND)
+
+        if host_request.status == HostRequestStatus.pending:
+            context.abort(grpc.StatusCode.FAILED_PRECONDITION, errors.HOST_REQUEST_PENDING_ARCHIVE_ATTEMPT)
```

Comment:
I think past requests can be pending, so maybe add a check here to check it's upcoming; or you can simply remove this check.